kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -651,45 +626,54 @@ class ResidualBasedBlockBuilderAndSolverWithConstraints
     {
         KRATOS_TRY
         int slave_count = 0;
-        for (auto &slave_equation_id : rSlaveEquationIdVector)
+        for (auto slave_equation_id : rSlaveEquationIdVector)
         {
+            /*int master_count = 0;
+            auto global_constraint = mGlobalMasterSlaveConstraints.find(slave_equation_id);
+            if (global_constraint == mGlobalMasterSlaveConstraints.end())
+            {
+                mGlobalMasterSlaveConstraints.insert(mGlobalMasterSlaveConstraints.begin(), Kratos::make_shared<AuxiliaryGlobalMasterSlaveConstraintType>(slave_equation_id));
+                global_constraint = mGlobalMasterSlaveConstraints.find(slave_equation_id);
+            }*/
             int master_count = 0;
-            auto global_constraint = mGlobalMasterSlaveRelations.find(slave_equation_id);
-            if (global_constraint == mGlobalMasterSlaveRelations.end())
+            if (mGlobalMasterSlaveConstraints.count(slave_equation_id) == 0 )
```

Comment:
this call is more expensive than seeing if it is not just there. I think that the previois version was more efficient